telegram/ext/updater.py

```diff
@@ -251,11 +255,54 @@ def start_polling(self,
         Returns:
             :obj:`Queue`: The update queue that can be filled from the main thread.
 
+        Raises:
+            ValueError: If :attr:`clean` is set as :obj:`datetime.timedelta` and is < 1 second.
+
         """
         with self.__lock:
             if not self.running:
                 self.running = True
 
+                self.logger.debug('clean:               "%s"', type(clean))
+                self.logger.debug('clean:               "%s"', clean)
+                self.logger.debug('now:                 "%s"', datetime.now(tz=timezone.utc))
+
+                if isinstance(clean, timedelta):
+                    if clean.total_seconds() < 1:
+                        raise ValueError('Clean as timedelta needs to be >= 1 second')
+                    else:
+                        a=(datetime.now(tz=timezone.utc) - clean)
+                        self.logger.debug('now - delta:         "%s"', a)
+                        self.logger.debug('now - delta type:    "%s"', type(a))
+                        self.logger.debug('clean as delta:      "%s"', clean)
+                        # convert to datetime
+                        clean = datetime.now(tz=timezone.utc) - clean
+                        self.logger.debug('clean as datetime:   "%s"', clean)
+                elif isinstance(clean, datetime):
+
+                    self.logger.debug('clean tz:            "%s"', clean.tzname())
+
+                    if (
+                            clean.tzinfo is None or
+                            (clean.tzinfo is not None and clean.tzinfo.utcoffset(clean) is None)
```

Comment:
What is the second check good for? If the datetime is aware it shouldn't matter what the UTC offset is?

Reply:
per [PY 3.6.10 documentation (scroll down just a little to the datetime.timezone section](https://docs.python.org/3.6/library/datetime.html?highlight=datetime#available-types) the requirement to be 'aware' is to have both conditions. If UTC offset is `None` datetime is naive.

[](url)
> An object of type time or datetime may be naive or aware. A datetime object d is aware if d.tzinfo is not None and d.tzinfo.utcoffset(d) does not return None. If d.tzinfo is None, or if d.tzinfo is not None but d.tzinfo.utcoffset(d) returns None, d is naive. A time object t is aware if t.tzinfo is not None and t.tzinfo.utcoffset(None) does not return None. Otherwise, t is naive.

Reply:
[PY 3.8.3 documentation](https://docs.python.org/3.8/library/datetime.html?highlight=datetime#determining-if-an-object-is-aware-or-naive) has even dedicated a paragraph on it

Reply:
I thought about changing it to
`clean.tzinfo.utcoffset(clean) is None`
but since the documentation is so specific on both requirements I don't think it's a good idea.

Reply:
We don't check `utcoffset` in other places, too… If people set non-working stuff as `tzinfo`, they must expect to get errors. Also the refactoring of `JobQueue` will make people use `pytz` with PTB anyway, so they might as well use it here